src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -1410,22 +1436,23 @@ public static boolean canBeSolved(Expression expr) {
 
   /**
    * This method takes a MethodCallExpr as an instance, and check if the method involved is called
-   * by an incomplete synthetic class. It should be noted that an incomplete synthetic class is
-   * different from a non-existing synthetic class. In this context, an incomplete synthetic class
-   * is a compilable class but missing some methods.
+   * by an incomplete synthetic class. An incomplete class could either be an original class with
+   * unsolved symbols or a synthetic class that need to be updated.
    *
    * @param method a MethodCallExpr instance
-   * @return true if the method involved is called by an incomplete synthetic class
+   * @return true if the method involved is called by an incomplete class
    */
-  public static boolean calledByAnIncompleteSyntheticClass(MethodCallExpr method) {
+  public static boolean calledByAnIncompleteClass(MethodCallExpr method) {
     if (calledByAnUnsolvedSymbol(method)) {
       return false;
     }
     if (method.getScope().isEmpty()) {
       return false;
     }
     try {
-      method.resolve();
+      // use an additional getReturnType() will check the solvability of the method more
+      // comprehensively.
```

Comment:
This comment is vague. Can you amend it to say in what situations this change is useful?